src/callbacks/solution_saving.jl

```diff
@@ -209,99 +209,75 @@ function (finalize::SolutionSavingCallback)(c, u, t, integrator)
     return nothing
 end
 
-function Base.show(io::IO, cb::DiscreteCallback{<:Any, <:SolutionSavingCallback})
-    @nospecialize cb # reduce precompilation time
-
-    solution_saving = cb.affect!
-    print(io, "SolutionSavingCallback(interval=", solution_saving.interval, ")")
-end
-
-function Base.show(io::IO,
-                   cb::DiscreteCallback{<:Any,
-                                        <:PeriodicCallbackAffect{<:SolutionSavingCallback}})
-    @nospecialize cb # reduce precompilation time
-
-    solution_saving = cb.affect!.affect!
-    print(io, "SolutionSavingCallback(dt=", solution_saving.interval, ")")
-end
-
-function Base.show(io::IO,
-                   cb::DiscreteCallback{<:Any, <:Any, <:Any, <:SolutionSavingCallback})
-    @nospecialize cb # reduce precompilation time
-
-    solution_saving = cb.finalize
-    print(io, "SolutionSavingCallback(save_times=", solution_saving.save_times, ")")
-end
-
-function Base.show(io::IO, ::MIME"text/plain",
-                   cb::DiscreteCallback{<:Any, <:SolutionSavingCallback})
-    @nospecialize cb # reduce precompilation time
-
-    if get(io, :compact, false)
-        show(io, cb)
+function Base.show(io::IO, cb::DiscreteCallback)
```

Comment:
In general, this might break other functionality, even of other packages, as core functionality of SciMLBase is overwritten.

Reply:
We never want to do type piracy, unless we have a very good reason to do so. In main and #669, all dispatches contain our custom type somewhere, so it's not type piracy.